From a5888efdc46bc083a460ed5be0196960c6a34f09 Mon Sep 17 00:00:00 2001
From: Tommy Huang <tommy_huang@aspeedtech.com>
Date: Mon, 22 Sep 2025 14:42:10 +0800
Subject: [PATCH 10/10] cmd: aspeedmbox: add ltpi channel info into misc

Assign the ltpi channel information into misc.

Signed-off-by: Tommy Huang <tommy_huang@aspeedtech.com>
Change-Id: I6ff6b9b3f46f62b526d6bef390014d07d39ec11f
(cherry picked from commit 9bd5b9afd4335e5164e8ff3a737af6af10b2f1b2)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 cmd/aspeed/aspeed_mbox.c | 30 ++++++++++++++++++++++++++----
 1 file changed, 26 insertions(+), 4 deletions(-)

diff --git a/cmd/aspeed/aspeed_mbox.c b/cmd/aspeed/aspeed_mbox.c
index a5d604c74a..3610467696 100644
--- a/cmd/aspeed/aspeed_mbox.c
+++ b/cmd/aspeed/aspeed_mbox.c
@@ -15,6 +15,8 @@
 #define LTPISHEADER		0xf0000000
 #define LTPIMISCREAD	0x0
 #define LTPIMISCWRITE	0x1
+#define LTPICHANNEL0	0x10000
+#define LTPICHANNEL1	0x20000
 
 enum misc_op {
 	MISC_OP_READ,
@@ -251,7 +253,7 @@ static int do_mbox_ltpi_read(struct cmd_tbl *cmdtp, int flag,
 	struct udevice *dev;
 	u32 *buf = NULL;
 	struct ltpi_ipc_data *data = NULL;
-	u32 msg[8];
+	u32 msg[8], misc = 0;
 	int ret = 0, i;
 	u32 channel = 0, offset, size, data_size;
 
@@ -276,10 +278,20 @@ static int do_mbox_ltpi_read(struct cmd_tbl *cmdtp, int flag,
 
 	data = (struct ltpi_ipc_data *)buf;
 
+	/* assign ltpi channel info */
+	if (channel == 0) {
+		misc |= LTPICHANNEL0;
+	} else if (channel == 1) {
+		misc |= LTPICHANNEL1;
+	} else {
+		printf("Not permitted access!\n");
+		return -EPERM;
+	}
+
 	/* fill read data */
 	for (i = 0; i < size ; i++) {
 		/* read the port data and save into value */
-		data->misc = LTPIMISCREAD;
+		data->misc = (misc | LTPIMISCREAD);
 		data->port = offset + (i << 2);
 		data->value = 0;
 
@@ -329,7 +341,7 @@ static int do_mbox_ltpi_write(struct cmd_tbl *cmdtp, int flag,
 	struct udevice *dev;
 	u32 *buf = NULL;
 	struct ltpi_ipc_data *data = NULL;
-	u32 msg[8];
+	u32 msg[8], misc = 0;
 	int ret = 0, i;
 	u32 channel = 0, offset, value, data_size;
 
@@ -354,10 +366,20 @@ static int do_mbox_ltpi_write(struct cmd_tbl *cmdtp, int flag,
 
 	data = (struct ltpi_ipc_data *)buf;
 
+	/* assign ltpi channel info */
+	if (channel == 0) {
+		misc |= LTPICHANNEL0;
+	} else if (channel == 1) {
+		misc |= LTPICHANNEL1;
+	} else {
+		printf("Not permitted access!\n");
+		return -EPERM;
+	}
+
 	/* fill read data */
 	for (i = 0; i < 1 ; i++) {
 		/* read the port data and save into value */
-		data->misc = LTPIMISCWRITE;
+		data->misc = (misc | LTPIMISCWRITE);
 		data->port = offset + (i << 2);
 		data->value = value;
 
-- 
2.34.1

